### PR TITLE
Support for the JSON.ARRPOP command

### LINKED ISSUE #1810: Add support for the JSON.ARRPOP command
Please refer to https://github.com/apache/kvrocks/issues/1805.

### PR DESCRIPTION
**Pull request by jyf111**
This PR adds support for the JSON.ARRPOP command.

closes #1810.

### CODE REVIEW

File: src/types/json.h [hunk 1]
```diff
@@ -215,6 +215,32 @@ struct JsonValue {
...
+    try {
+      jsoncons::jsonpath::json_replace(value, path,
+                                       [&popped_values, index](const std::string & /*path*/, jsoncons::json &val) {
+                                         if (val.is_array() && !val.empty()) {
+                                           auto popped_iter = val.array_range().begin();
+                                           if (index == -1 || index >= static_cast<int64_t>(val.size())) {
+                                             popped_iter = val.array_range().end() - 1;
+                                           } else if (index > 0) {
+                                             popped_iter += index;
+                                           }
```
Comment (PragmaTwice) [inline src/types/json.h]: I think if the index is negative, we need to calculate the real position from the end. 
```shell
127.0.0.1:6666> json.set a $ [1,2,3,4]
OK
127.0.0.1:6666> json.arrpop a $ -2
1) "3"
127.0.0.1:6666>
```
I recommend you to try it by yourself : )

Comment (jyf111) [reply src/types/json.h]: I'm sorry I misunderstood `Out-of-range indexes round to their respective array ends` stated in the doc.

I've fixed it based on the below code in RedisJson.

```rust
impl ArrayIndex for i64 {
    fn normalize(self, len: i64) -> usize {
        let index = if self < 0 {
            len - len.min(-self)
        } else if len > 0 {
            (len - 1).min(self)
        } else {
            0
        };
        index as usize
    }
}
```

**Comment by PragmaTwice** [comment]
Could you try to fix these issue reported by clang-tidy in CI?

https://github.com/apache/kvrocks/actions/runs/6754872818/job/18363385655?pr=1874

**Comment by jyf111** [comment]
> Could you try to fix these issue reported by clang-tidy in CI?

Fixed.

**Review by PragmaTwice — APPROVED** [review]

**Comment by PragmaTwice** [comment]
Thank you for your contribution!

**Pull request merged**